lib/app/features/feed/create_post/views/pages/create_post_modal/create_post_modal.dart

```diff
@@ -38,9 +45,75 @@ class CreatePostModal extends HookWidget {
   }
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
     final textEditorController = useQuillController();
     final hasContent = useTextEditorHasContent(textEditorController);
+    final hasPoll = useState(false);
+
+    useEffect(
+      () {
+        void checkPoll() {
+          final delta = textEditorController.document.toDelta();
+          var pollExists = false;
+
+          for (final operation in delta.operations) {
+            if (operation.isInsert && operation.data is Map<String, dynamic>) {
```

Comment:
Could you consider using `records` ([link](https://dart.dev/language/records#multiple-returns)) and `patterns` ([link](https://dart.dev/language/patterns#validating-incoming-json)) to simplify this code?

Reply:
it's optional, up to you